bindings/py/cpp_src/bindings/algorithms/py_SpatialPooler.cpp

```diff
@@ -342,16 +342,17 @@ Argument output An SDR representing the winning columns after
         });
 
         // getPermanence
-        py_SpatialPooler.def("getPermanence", [](const SpatialPooler& self, UInt column, py::array& x)
+        py_SpatialPooler.def("getPermanence", [](const SpatialPooler& self, const UInt column, py::array& x, const Permanence threshold = 0.0f)
         {
-            self.getPermanence(column, get_it<Real>(x));
+            const auto& perm = self.getPermanence(column, threshold);
+	    std::copy(perm.begin(), perm.end(), get_it<Real>(x)); //TODO pass-by-value here only for compatibility, could have just returned perm 
```

Comment:
I have some problem with the py-test, it used to check for some invalid dtype combinations. I don't know if that's still concern. Or if I should break compatibility, remove the argument and just return?